soroban-env-host/src/host.rs

```diff
@@ -3264,6 +3264,106 @@ impl VmCallerEnv for Host {
         self.bn254_check_pairing_output(&output)
     }
 
+    fn poseidon_permutation(
```

Comment:
It doesn't seem like the host functions are covered by the tests. I also haven't found any tests for invalid arguments, and I think it would make sense to test those via the public (host function) interface.